### PR TITLE
Add new package for serialization/deserialization and create enum for …

### PR DESCRIPTION
**Pull request by rdykiel**
…encoding formats. Use encoding format in DDL checking

### CODE REVIEW

File: src/frontend/org/voltdb/compiler/DDLCompiler.java [hunk 1]
```diff
@@ -1643,12 +1639,16 @@ private static void addTopicToCatalogTable (Table table,
         }
         String topicFormatName = node.attributes.get(SQLParser.CAPTURE_TOPIC_FORMAT);
         if (topicFormatName != null) {
-            if (!s_topicFormats.contains(topicFormatName)) {
+            try {
+                // Parse the format in an unchecked fashion and handle exceptions
+                EncodeFormat format = Enum.valueOf(EncodeFormat.class, topicFormatName);
```
Comment (n3world) [inline src/frontend/org/voltdb/compiler/DDLCompiler.java]: Minor: You can also do `format = EncodeFormat.valueOf(topicFormatName)`

File: src/frontend/org/voltdb/serdes/EncodeFormat.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    /**
+     * Parse an {@link EncodeFormat} from a {@link String} whose
+     * value is assumed to have been checked beforehand.
+     *
+     * @param name
+     * @return {@link EncodeFormat} value
+     */
+    public static EncodeFormat checkedValueOf(String name) {
+        try {
+            return Enum.valueOf(EncodeFormat.class, name);
```
Comment (n3world) [inline src/frontend/org/voltdb/serdes/EncodeFormat.java]: Why not use `EncodeFormat.valueOf`

File: src/frontend/org/voltdb/compiler/DDLCompiler.java [hunk 3]
```diff
@@ -1643,12 +1639,16 @@ private static void addTopicToCatalogTable (Table table,
         }
         String topicFormatName = node.attributes.get(SQLParser.CAPTURE_TOPIC_FORMAT);
         if (topicFormatName != null) {
```
Comment (n3world) [inline src/frontend/org/voltdb/compiler/DDLCompiler.java]: If it is null shouldn't this also the default format

File: src/frontend/org/voltdb/compiler/DDLCompiler.java [hunk 4]
```diff
@@ -1643,12 +1639,16 @@ private static void addTopicToCatalogTable (Table table,
...
-            if (!s_topicFormats.contains(topicFormatName)) {
+            try {
+                // Parse the format in an unchecked fashion and handle exceptions
+                EncodeFormat format = Enum.valueOf(EncodeFormat.class, topicFormatName);
+                table.setTopicformat(format.toString());
```
Comment (n3world) [inline src/frontend/org/voltdb/compiler/DDLCompiler.java]: I would recommend using format.name() if you are going to use valueOf to convert a string which is the name of the enum back to the enum.

File: src/frontend/org/voltdb/compiler/DDLCompiler.java (hunk 3 above)
Comment (rdykiel) [reply src/frontend/org/voltdb/compiler/DDLCompiler.java]: The determination of the default format is DEFERRED to the data source

**Pull request merged**